api_v3/lib/types/entry/filters/KalturaBaseEntryFilter.php

```diff
@@ -195,36 +195,34 @@ protected function doGetListResponse(KalturaFilterPager $pager)
 		myDbHelper::$use_alternative_con = myDbHelper::DB_HELPER_CONN_PROPEL3;
 
 		$disableWidgetSessionFilters = false;
-		if ($this &&
-			($this->idEqual != null ||
-			$this->idIn != null ||
-			$this->referenceIdEqual != null ||
-			$this->redirectFromEntryId != null ||
-			$this->referenceIdIn != null || 
-			$this->parentEntryIdEqual != null))
-			$disableWidgetSessionFilters = true;
-			
 		$c = $this->prepareEntriesCriteriaFilter($pager);
-		
-		if ($disableWidgetSessionFilters)
+		if ($this && ($this->idEqual != null || $this->idIn != null || $this->referenceIdEqual != null ||
+			$this->redirectFromEntryId != null || $this->referenceIdIn != null || $this->parentEntryIdEqual != null))
```

Comment:
Validating the values against != null misses values such as ''.
Better validate if ($this->idEqual  || $this->idIn ||. ...)